core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -70,83 +210,147 @@ sealed abstract class Pull[+F[_], +O, +R] {
 
   /** Interpret this `Pull` to produce a `Stream` without introducing a scope.
     *
-    * Only use this if you know a scope is not needed. Scope introduction is generally harmless and the risk
-    * of not introducing a scope is a memory leak in streams that otherwise would execute in constant memory.
+    * Only use this if you know a scope is not needed. Scope introduction is 
+    * generally harmless and the risk of not introducing a scope is a memory leak
+    * in streams that otherwise would execute in constant memory.
     *
-    * May only be called on pulls which return a `Unit` result type. Use `p.void.stream` to explicitly
+    * May only be called on closed pulls. Use `p.void.stream` to explicitly 
     * ignore the result type of the pull.
     */
   def streamNoScope(implicit ev: R <:< Unit): Stream[F, O] = {
     val _ = ev
     new Stream(this.asInstanceOf[Pull[F, O, Unit]])
   }
 
-  /** Applies the resource of this pull to `f` and returns the result. */
-  def flatMap[F2[x] >: F[x], O2 >: O, R2](f: R => Pull[F2, O2, R2]): Pull[F2, O2, R2] =
-    new Bind[F2, O2, R, R2](this) {
-      def cont(e: Terminal[R]): Pull[F2, O2, R2] =
-        e match {
-          case Succeeded(r) =>
-            try f(r)
+  /** Attaches to `this` pull a handler for errors that may be raised from it.
```

Comment:
Reword to something like:

```
Allows recovery from an error that occurs during evaluation of this pull.

If an error occurs, the supplied function is called to determine how to continue processing. Note that the original computation cannot be resumed at the point of an error.
```

Then provide an example using `@example`.

I'd avoid describing continuation pulls and compilation semantics.